### ISSUE TITLE
Replying to comments

### ISSUE THREAD

**Issue by bates64** [OpenSprites/OpenSprites-next#93]
~~Comments should be 1-tier repliable, similar to Scratch.~~

After discussion, multiple tiers are allowed, but with the `@mention`-style that Scratch uses.

**Comment by towerofnix** [comment]
But

> everyone
> 
> > likes
> > 
> > > doing
> > > 
> > > > this!

📦

**Comment by Tymewalk** [comment]
So the reply buttons on subcomments have to link back to the root comment?

**Comment by towerofnix** [comment]
It would probably be better to have a box (or button that shows the comment leaving dialog) to leave a comment in at the bottom of a comment chain, rather than "reply"; unless reply added a quote to the comment box, which might be nice.

**Comment by bates64** [comment]
Not sure about quotes- they're only comments at the end of the day.

**Comment by Tymewalk** [comment]
@liam4 Reply could append an "@username" to the beginning of a message. Like it does on Scratch. Plus it would notify that user and not the owner of the root comment.

**Comment by bates64** [comment]
We should make _any_ instance of "@username" notify the user.

**Comment by towerofnix** [comment]
> Reply could append an "@username" to the beginning of a message. Like it does on Scratch. Plus it would notify that user and not the owner of the root comment.

@Tymewalk Shouldn't it append to the end of the message rather than the beginning? Since you'll be writing that part of the reply after.

**Comment by bates64** [comment]
> > Reply could append an "@username" to the beginning of a message. Like it does on Scratch. Plus it would notify that user and not the owner of the root comment.
> 
> @Tymewalk Shouldn't it append to the end of the message rather than the beginning? Since you'll be writing that part of the reply after.

You guys are going on about `append` ing to the beginning of messages and stuff, like, [do you not know what "append" means?](https://www.google.co.uk/?q=append%20definition)

TL;DR: "append" means add to the _end_ wheras "prepend" means add at the _start_.
Get it right, people. :shipit:

**Comment by Tymewalk** [comment]
@nanalan Where's the facedesk emoji? I need 7 of them.

Reply would ~~add~~ prepend the "@username" to the ~~beginning of the~~ message.

For example:

```
Hey there guys (Tymewalk)
\ @Tymewalk What's up? (nanalan)
| \ @nanalan Not much, being a theoretical example (Tymewalk)
\ (more comments here)
```

**Comment by bates64** [comment]
FTFY @Tymewalk
